### DISCUSSION TITLE
Publish and Deploy .Net WebAPI in MacOS locally

### DISCUSSION THREAD

**Discussion by PD0704** [microsoft/vscode-discussions#1087] [Extension Development QnA]
Hello, this is first time I have created a discussion thread in Github, so I apologise for any mistake I do.
As for the question,
I have created a WEB API project and want to deploy it locally and I am using MacOS. I have found solution regarding Windows which is IIS and Nginx for Linux but not able to find proper solution for MacOS. 
So, little guidance will be helpful.
Thank You.

**Comment by abrilurena** [comment]
Hello! No need to apologize—I’m here to help! 😊 I found this, please let me know if this helps you. 

1. Run from Visual Studio (Development Mode):

- If you’re using Visual Studio, you can simply run your ASP.NET Core Web API project in development mode by pressing F5. This will start the application, and you can access it from your browser.
- Visual Studio will automatically launch a browser window and navigate to the appropriate URL where your site is hosted.

2. Run from Command Prompt:

- Open a terminal or command prompt window in the root folder of your ASP.NET Core Web API project.
- Use the dotnet run command to run your application. This will start the API on a specified port (usually 5000 or 5001).
- You’ll need to configure your consuming application (like your Flutter app) to use the correct port.

3. Host Locally with IIS (Windows OS):

- If you’re running macOS but have access to a Windows machine, you can set up IIS (Internet Information Services) to host your ASP.NET Core Web API.
- Follow the instructions in this tutorial: Host ASP.NET Core in IIS.
- Remember that when running locally, your API will be accessible at http://localhost:port. Make sure to configure your Flutter app to make requests to this URL.

Feel free to choose the option that best suits your needs. If you have any more questions, feel free to ask! 😊

**Reply by PrahelikaD** [reply]
I am currently working in VS Code instead of Visual Studio as Visual Studio is going to stop working for MacOS. And for the UI or frontend part, I am using Unity App.

So, will this steps be similar as you have mentioned above? Or do I have to change few steps for this?

**Reply by abrilurena** [reply]
Hello @PrahelikaD! I'm so sorry, I got confused, however I found these solutions, please let me know if this helps you:

- [Live Server extension](https://marketplace.visualstudio.com/items?itemName=ritwickdey.LiveServer&WT.mc_id=academic-00000-abrilu) - i use it sometimes when i want to launch something locally but I'm not sure if it would work for an API. 
- [Tutorial: Create a web API with ASP.NET Core](https://learn.microsoft.com/en-us/aspnet/core/tutorials/first-web-api?view=aspnetcore-8.0&tabs=visual-studio-code&WT.mc_id=academic-00000-abrilu) - I saw there are some steps for MacOS :D 
- [Developing ASP.NET Core Applications on a Mac With Visual Studio Code](https://aspnetcore.readthedocs.io/en/stable/tutorials/your-first-mac-aspnet.html#id3)


In case none of these solutions works, it's totally okaaay, just let me know so I'll include someone from the team that can help you. ✨

**Comment by burkeholland** [comment]
This doesn't appear to be on topic for extension authoring - it's probably better suited for a forum like Stack Overflow. Closing for now, but feel free to-reopen if you feel this is an error.
